src/hooks/usePhantomWallet.ts

```diff
@@ -0,0 +1,41 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { PublicKey } from '@solana/web3.js';
+
+export function usePhantomWallet() {
+  const [solAddress, setSolAddress] = useState('');
+
+  const connect = useCallback(async (): Promise<string> => {
```

Comment:
We'll also need to add `disconnect` to this hook.

Reply:
✅